MetroIntentsUI/IntentViewController.swift

```diff
@@ -0,0 +1,61 @@
+import IntentsUI
+import SwiftUI
+import MetroKit
+import MetroUI
+
+class IntentViewController: UIViewController, INUIHostedViewControlling {
+    let padding: CGFloat = 8
+
+    // MARK: - INUIHostedViewControlling
+
+    var desiredSize: CGSize {
+        let maxWidth = self.extensionContext!.hostedViewMaximumAllowedSize.width
+        return CGSize(width: maxWidth, height: maxWidth / MetroCardView.aspectRatioMultiplier + padding * 2)
+    }
+
+    // Prepare your view controller for the interaction to handle.
+    func configureView(for parameters: Set<INParameter>, of interaction: INInteraction, interactiveBehavior: INUIInteractiveBehavior, context: INUIHostedViewContext, completion: @escaping (Bool, Set<INParameter>, CGSize) -> Void) {
+        func completeWithoutUI() {
+            completion(false, parameters, .zero)
+        }
+
+        // Handle supported responses
+        guard
+            let response = interaction.intentResponse as? BalanceProvidingIntentResponse,
+            let balance = response.balance,
+            let currencyCode = balance.currencyCode,
+            let amount = balance.amount
+        else {
+            return completeWithoutUI()
+        }
+
+        let numberFormatter = NumberFormatter()
+        numberFormatter.currencyCode = currencyCode
+        numberFormatter.numberStyle = .currency
+
+        guard let formattedBalance = numberFormatter.string(from: amount) else {
+            return completeWithoutUI()
+        }
+
+        addResponseViewController(formattedBalance: formattedBalance)
+        completion(true, parameters, desiredSize)
+    }
+
+    private func addResponseViewController(formattedBalance: String) {
+        let rootView = IntentResponseView(formattedBalance: formattedBalance)
+
+        let hostingController = UIHostingController(rootView: rootView)
+        hostingController.willMove(toParent: self)
+        addChild(hostingController)
```

Comment:
`addChild()` automatically calls `willMove(toParent:)`, only `didMove(toParent:)` is explicitly necessary.